src/riak_kv_compile_tab.erl

```diff
@@ -151,7 +153,87 @@ get_ddl(BucketType, Version) when is_binary(BucketType), is_atom(Version) ->
             notfound
     end.
 
-%%
+%% Get the list of {TableName, Status} pairs, no matter what status they are in.
+-spec get_table_status_pairs() ->[{binary(), binary()}].
+get_table_status_pairs() ->
+    Matches = dets:match(?TABLE3, #row_v3{ table = '$1' }),
+    UTables = lists:usort(Matches),
+    Tables = [ {Table, get_table_status(Table)} ||
+               [Table|_T] <- UTables ],
+
+    %% sort by Status (thankfully in alphabetic sort order):
+    %% Active, Not Active, Undefined
+    SortByStatus = fun({L1,L2},{R1,R2}) -> {L2,L1} =< {R2,R1} end,
+    lists:sort(SortByStatus, Tables).
+
+get_table_status(Table) ->
+    case get_table_ddl(Table) of
+        {error, no_type} -> <<"Not Active">>;
+        {error, missing_helper_module} -> <<"Not Active">>;
+        {ok, _Mod, _DDL} -> get_table_status_by_version(Table);
+        _ -> get_table_status_by_version(Table)
+    end.
+
+get_table_status_by_version(Table) ->
+    Req = #tsqueryreq{},
+    DecodedReq = {ok, Req, {"perm", Table}},
+    case check_table_feature_supported(v2, DecodedReq) of
+        DecodedReq -> <<"Active">>;
+        _ -> <<"Not Active">>
+    end.
+
+%% Forwards/Mocks for getting table status, isolating the interaction for testability
+-ifndef(TEST).
+-type ts_service_req() ::
+    {ok, riak_kv_ts_svc:ts_requests(), {PermSpec::string(), Table::binary()}}.
+
+-spec get_table_ddl(binary()) ->
+    {ok, module(), ?DDL{}} | {error, term()}.
+get_table_ddl(Table) ->
+    riak_kv_ts_util:get_table_ddl(Table).
+
+-spec check_table_feature_supported(DDLRecCap::atom(),
+                                    DecodedReq::ts_service_req()) -> ts_service_req() |
+                                                                     {error, string()}.
+check_table_feature_supported(DDLRecCap, DecodedReq) ->
+    riak_kv_ts_util:check_table_feature_supported(DDLRecCap, DecodedReq).
+
+-else.
+expected_table_status(Table) when Table =:= <<"my_type2">> ->
+    <<"Not Active">>;
+expected_table_status(Table) when Table =:= <<"my_type4">> ->
+    <<"Not Active">>;
+expected_table_status(Table) when Table =:= <<"my_type5">> ->
+    <<"Not Active">>;
+expected_table_status(_Table) ->
+    <<"Active">>.
+
+get_table_ddl(Table) when Table =:= <<"my_type4">> ->
+    {error, no_type};
+get_table_ddl(Table) when Table =:= <<"my_type5">> ->
+    {error, missing_helper_module};
+get_table_ddl(_Table) ->
+    Module = {}, %% not used by caller
+    DDL = {}, %% not used by caller
+    {ok, Module, DDL}.
+
+check_table_feature_supported(DDLRecCap, DecodedReq={ok, _Req, {"perm", Table}}) when
```

Comment:
The conditions required to be active are greater than the conditions required to not be, so I posed the function guards along those lines. Being pessimistic in this way IMHO is more than just "future proofing" but should make the coding of expected results easier to reason about.